lib/class/api.class.php

```diff
@@ -724,26 +1069,55 @@ public static function advanced_search($input)
     {
         ob_end_clean();
 
-        XML_Data::set_offset($input['offset']);
-        XML_Data::set_limit($input['limit']);
+        //Whatever format the user wants
+        $outputFormat = $input['format'];
+
+        if ($outputFormat == 'json') {
+            JSON_Data::set_offset($input['offset']);
+            JSON_Data::set_limit($input['limit']);
 
-        $results = Search::run($input);
+            $results = Search::run($input);
```

Comment:
Same, most of these lines should go out of the test case, they are not format-specific.